### PR TITLE
Fix limit on detail twin masters

### PR DESCRIPTION
**Pull request by bratchikov**

### CODE REVIEW

**Comment by bratchikov** [comment]
This method need to be fixed for this case: https://github.com/Flexberry/NewPlatform.Flexberry.ORM.ODataService/blob/1e87cec96b7d43be76c7b6e1442ed23e3a14bec6/NewPlatform.Flexberry.ORM.ODataService/Expressions/FilterBinder.cs#L2086

File: Tests/NewPlatform.Flexberry.ORM.ODataService.Tests/CRUD/Read/FilterByMasterMasterDetailFieldTest.cs [hunk 1]
```diff
@@ -1,4 +1,4 @@
-﻿namespace NewPlatform.Flexberry.ORM.ODataService.Tests.CRUD.Read
+﻿ namespace NewPlatform.Flexberry.ORM.ODataService.Tests.CRUD.Read
```
Comment (NicholasNoise) [inline Tests/NewPlatform.Flexberry.ORM.ODataService.Tests/CRUD/Read/FilterByMasterMasterDetailFieldTest.cs:1]:
```suggestion
﻿namespace NewPlatform.Flexberry.ORM.ODataService.Tests.CRUD.Read
```

File: NewPlatform.Flexberry.ORM.ODataService/Expressions/FilterBinder.cs [hunk 2]
```diff
@@ -2102,37 +2102,64 @@ private void FillFilterDetailProperties(LambdaNode anyNode)
...
+            {
+                return;
+            }
+
+            SingleValuePropertyAccessNode[] propertyNodes = new SingleValuePropertyAccessNode[0];
+            var binaryNode = node as BinaryOperatorNode;
+
+            if (binaryNode != null)
+            {
+                if (binaryNode.Left is SingleValuePropertyAccessNode)
```
Comment (NicholasNoise) [inline NewPlatform.Flexberry.ORM.ODataService/Expressions/FilterBinder.cs:2122]: Should we consider same for `binaryNode.Right`?

File: NewPlatform.Flexberry.ORM.ODataService/Expressions/FilterBinder.cs [hunk 3]
```diff
@@ -2102,37 +2102,64 @@ private void FillFilterDetailProperties(LambdaNode anyNode)
...
+        {
+            if (node == null)
+            {
+                return;
+            }
+
+            SingleValuePropertyAccessNode[] propertyNodes = new SingleValuePropertyAccessNode[0];
+            var binaryNode = node as BinaryOperatorNode;
+
+            if (binaryNode != null)
```
Comment (NicholasNoise) [inline NewPlatform.Flexberry.ORM.ODataService/Expressions/FilterBinder.cs:2120]:
```suggestion
            if (node is BinaryOperatorNode binaryNode)
```

File: NewPlatform.Flexberry.ORM.ODataService/Expressions/FilterBinder.cs [hunk 4]
```diff
@@ -2102,37 +2102,64 @@ private void FillFilterDetailProperties(LambdaNode anyNode)
...
-                        propertyNodes = new SingleValuePropertyAccessNode[] { binaryNode.Left as SingleValuePropertyAccessNode, binaryNode.Right as SingleValuePropertyAccessNode };
+                        GetFilterDetailPropertiesFromNode(binaryNode.Right, navigationPropertyName, filterDetailProperties);
                     }
+                }
+            }
 
-                    var functionCallNode = anyNode.Body as SingleValueFunctionCallNode;
-                    if (functionCallNode != null && functionCallNode.Name == "contains")
+            var functionCallNode = node as SingleValueFunctionCallNode;
+            if (functionCallNode != null && functionCallNode.Name == "contains")
```
Comment (NicholasNoise) [inline NewPlatform.Flexberry.ORM.ODataService/Expressions/FilterBinder.cs:2138]:
```suggestion
            if (functionCallNode.Name == "contains" && node is SingleValueFunctionCallNode functionCallNode)
```

File: NewPlatform.Flexberry.ORM.ODataService/Expressions/FilterBinder.cs [hunk 5]
```diff
@@ -2102,37 +2102,64 @@ private void FillFilterDetailProperties(LambdaNode anyNode)
...
+            {
+                var parameters = functionCallNode.Parameters.ToList();
```
Comment (NicholasNoise) [inline NewPlatform.Flexberry.ORM.ODataService/Expressions/FilterBinder.cs:2140]: What about this approach?
```suggestion
                propertyNodes = functionCallNode.Parameters.OfType<SingleValuePropertyAccessNode>().ToList();
```

**Pull request merged**